File: Algorithm/BruteForce/2798.swift
import Foundation

let NM = readLine()!.split(separator: " ").map { Int($0)! }
let cards = readLine()!.split(separator: " ").map { Int($0)! }

print(solution(NM[0], NM[1], cards))

func solution(_ N: Int, _ M: Int, _ cards: [Int]) -> Int {
  var answer: Int = 0

  for i in 0..<N {
    for j in i+1..<N {
      for k in j+1..<N {
        if cards[i]+cards[j]+cards[k] <= M {
          answer = max(answer, cards[i]+cards[j]+cards[k])
        }
      }
    }
  }

  return answer
}

File: Algorithm/BruteForce/2798.py
def solution(N, M, Cards):
    ret = 0

    for i in range(N-2):
        for j in range(i+1, N-1):
            for k in range(j+1, N):
                if M >= Cards[i]+Cards[j]+Cards[k]:
                    ret = max(ret, Cards[i]+Cards[j]+Cards[k])

    return ret


N, M = map(int, input().split())
Cards = list(map(int, input().split()))

print(solution(N, M, Cards))